src/api-transition-helpers.coffee

```diff
@@ -0,0 +1,17 @@
+LineNumberGutterName = 'line-numbers'
+
+# Given decorationParams/Properties from the 1.0 API, applies changes to enable
+# the 2.0 API under the covers. In 2.0, the `trueType` property will become `type`.
+translateDecorationParamsOldToNew = (decorationParams) ->
```

Comment:
Yeah, I think I like option 2, which is that we continue to treat `line-number` decorations in an idiosyncratic way. They are different from the other `gutter` decorations. I think it's the only path that doesn't require us to double back on our API commitments. I also don't think it's that crazy. Want to decorate a line number? Use a `line-number` decoration. Want to decorate some other gutter? Use a `gutter` decoration. I don't think it's that confusing.